version-catalog/src/main/kotlin/io/spine/internal/catalog/entry/AbstractVersionInheritingEntry.kt

```diff
@@ -0,0 +1,138 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.catalog.entry
+
+import io.spine.internal.catalog.Alias
+import io.spine.internal.catalog.CatalogRecord
+import io.spine.internal.catalog.VersionNotation
+import io.spine.internal.catalog.VersionRecord
+
+/**
+ * A skeleton implementation of a version entry, which can take the version
+ * from the outer entry.
+ *
+ * From the outside it looks like the entry just "inherits" the version from
+ * its parent.
+ *
+ * In order to understand why it's needed, consider the following snippet:
+ *
+ * ```
+ * internal object Kotlin : LibraryEntry() {
+ *     override val version = "1.6.21"          // libs.versions.kotlin
+ *     override val module = "jetbrains:kotlin" // libs.kotlin
+ *
+ *     object Runtime : LibraryEntry() {
+ *         override val version = Kotlin.version            // libs.versions.kotlin.runtime
+ *         override val module = "jetbrains:kotlin-runtime" // libs.kotlin.runtime
+ *     }
+ * }
+ * ```
+ *
+ * Both `Kotlin` and `Kotlin.Runtime` libraries will receive the same version,
+ * declared in a single place. But for each library, an independent version
+ * record will be created:
+ *
+ *  1. `libs.versions.kotlin`.
+ *  2. `libs.versions.kotlin.runtime`.
+ *
+ * Thus, a local overriding (in settings file) of `libs.versions.kotlin` will
+ * not affect the version of `Kotlin.Runtime` library. But, intuitively should.
+ *
+ * In contrast, when using an entry which extends this skeleton, there's no
+ * need in manual declaring the version in the nested entry. A version inheriting
+ * entry can take the version from the outer entry on its own.
+ *
+ * Consider the same snippet, but with an entry which extends this class:
+ *
+ * ```
+ * internal object Kotlin : LibraryEntry() {
+ *     override val version = "1.6.21"              // libs.versions.kotlin
+ *     override val module = "org.jetbrains:kotlin" // libs.kotlin
+ *
+ *     object Runtime : SomeVersionInheritingEntry() {
+ *         override val module = "org.jetbrains:kotlin-runtime" // libs.kotlin.runtime
+ *     }
+ * }
+ * ```
+ *
+ * Going this way, only `libs.versions.kotlin` will be generated and available
+ * for a local override. Both `Kotlin` and `Kotlin.Runtime` libraries will
+ * reference `libs.versions.kotlin` version. When this version is overridden,
+ * both libraries will be affected as well.
+ *
+ * Although, such entries are not bound to use only an inherited version.
+ * It is still possible to declare the version within this entry. In case,
+ * when neither this entry nor its outer one declares a version, an exception
+ * will be thrown.
+ */
+internal open class AbstractVersionInheritingEntry : CatalogEntry(), VersionNotation {
```

Comment:
I would name it after its structural features, not after its functional detail. E.g. `AbstractNestedEntry`.

The idea is that such nested objects may eventually have more functional features (as they already inherit versions), but they will always be a part of some bigger entry (such as `LibraryEntry`).